File: C++/0044. Calculate the Sum.cpp
// we have defined the necessary header files here for this problem.
// If additional header files are needed in your program, please import here.
#include <vector>
int main()
{  
  // please define the C++ input here. For example: int a,b; cin>>a>>b;;  
  // please finish the function body here.  
  // please define the C++ output here. For example:cout<<____<<endl; 

    int test;
    string m,n;
    int res;
    cin>>test;
    for(int _=0;_<test;++_){
        cin>>m>>n;
        res = 0;
        vector<int> arr(10,0);
        for(auto c:n){
            arr[c-'0']++;
        }
        for(auto c:m){
            for(int i=1;i<10;++i){
                res += (c-'0')%i*arr[i];
            }
        }
        cout<<res<<endl;
    }
    return 0;
}

File: Python/0044. Calculate the Sum.py
# If you need to import additional packages or classes, please import here.
import collections
def func():
    # please define the python3 input here. 
    # For example: a,b = map(int, input().strip().split())
    # please finish the function body here.
    # please define the python3 output here. For example: print().

    while True:
        try:
            test = int(input())
            for _ in range(test):
                res = 0
                m = input().strip()
                n = input().strip()
                dict_n = collections.defaultdict(int)
                for c in n:
                    dict_n[c] += 1

                for c in m:
                    for i in dict_n:
                        res += (int(c)%int(i))*dict_n[i]
                print(res)

        except EOFError:
            break
        
if __name__ == "__main__":
    func()
